internal/util/sort.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2024 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"cmp"
+	"sort"
+)
+
+// Sorting ensures, that we always generate the config in the same order.
+func SortedKeys[Key cmp.Ordered, Value any](input map[Key]Value) (keys []Key) {
+	for k := range input {
+		keys = append(keys, k)
+	}
+	sort.Slice(keys, func(i, j int) bool { return keys[i] < keys[j] })
+
+	return
+}
```

Comment:
(nit) I'm not a big fan of named return values. We can also use the slices.Sort() function (which is more compact).
```suggestion
func SortedKeys[Key cmp.Ordered, Value any](m map[Key]Value) []Key {
	keys := make([]Key, len(input))
	for k := range m {
		keys = append(keys, k)
	}
	slices.Sort(keys)

	return keys
}
```

Reply:
Thanks. Got it. 
Change it done.